### PR TITLE
engine: 24.0.4 release notes

### PR DESCRIPTION
**Pull request by neersighted**

### CODE REVIEW

**Review by dvdksn — DISMISSED** [review]

File: engine/release-notes/24.0.md [hunk 1]
```diff
@@ -19,6 +19,20 @@ For more information about:
...
+{% include release-date.html date="2023-07-07" %}
+
+For a full list of pull requests and changes in this release, refer to the relevant GitHub milestones:
+
+- [docker/cli, 24.0.4 milestone](https://github.com/docker/cli/issues?q=is%3Aclosed+milestone%3A24.0.4)
+- [moby/moby, 24.0.4 milestone](https://github.com/moby/moby/issues?q=is%3Aclosed+milestone%3A24.0.4)
+
+### Bug fixes and enhancements
+
+* Fix a panic during live-restore of containers with bind mounts. [moby/moby#45903](https://github.com/moby/moby/pull/45903)
```
Comment (thaJeztah) [inline engine/release-notes/24.0.md:33]: Do we need to mention that it's was a regression in 24.0.3?

Comment (neersighted) [reply engine/release-notes/24.0.md:33]: cc @dvdksn; I have no strong opinion on that myself.

Comment (dvdksn) [reply engine/release-notes/24.0.md:33]: Let's do it, helps explain the sudden release :)

Comment (neersighted) [reply engine/release-notes/24.0.md:33]: How's something like:
```suggestion
* Fix a regression introduced by 24.0.3: a panic during live-restore of containers with bind mounts. [moby/moby#45903](https://github.com/moby/moby/pull/45903)
```

Comment (neersighted) [reply engine/release-notes/24.0.md:33]: Or:
```suggestion
* Fix a panic during live-restore of containers with bind mounts (regression introduced in 24.0.3). [moby/moby#45903](https://github.com/moby/moby/pull/45903)
```

Comment (thaJeztah) [reply engine/release-notes/24.0.md:33]: last one seems fine.

first one I'd have written as "Fix  a regression introduced in 24.0.3 that cause a panic ..."

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**